### PR TITLE
Adding flink 12 versions and spark 3.0.3

### PR DESCRIPTION
**Pull request by ChethanUK**

### CODE REVIEW

**Comment by ChethanUK** [comment]
@marc0der Can you check this PR - above error is for SAP

**Comment by marc0der** [comment]
@ChethanUK feel free to remove any obsolete migrations if they cause the build to fail.

**Comment by ChethanUK** [comment]
HI, this seems to be working I mean if you click you can download. Don't know why it says obsolete..
```Error: Exception in thread "main" com.github.mongobee.exception.MongobeeException: Invalid url: https://github.com/SAP/SapMachine/releases/download/sapmachine-12.0.1/sapmachine-jdk-12.0.1_osx-x64_bin.tar.gz```

**Comment by ChethanUK** [comment]
@marc0der Can you review and merge.

**Comment by marc0der** [comment]
@ChethanUK Happy for you to remove anything to get your build green. All JDKs are now provided by DISCO in anycase, so they are all redundant.

**Comment by ChethanUK** [comment]
@marc0der Can you review and merge

**Review by marc0der — APPROVED** [review]
LGTM!

**Comment by marc0der** [comment]
Thanks for contributing!

**Comment by marc0der** [comment]
@ChethanUK after hitting the merge button I realised that you aren't setting the latest versions as default. Would you please be able to do that in a follow-up PR?

File: src/main/scala/io/sdkman/changelogs/SparkMigrations.scala
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/SparkMigrations.scala:246]: It seems that this is for Spark 3.0.3, not 3.1.1. It has already been merged but I'll back this one out.

File: src/main/scala/io/sdkman/changelogs/FlinkMigrations.scala
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/FlinkMigrations.scala:131]: I've had to back out all the versions that created duplicates in our database. Your code inserted, for instance, a `1.14.0` twice, once each for Scala 2.11 and 2.12. I've gone and removed all such duplicates as they break our CLI. I've only left the Scala `2.12` variants.

**Pull request merged**
